sklearn/discriminant_analysis.py

```diff
@@ -470,65 +483,68 @@ def _solve_svd(self, X, y):
         y : array-like of shape (n_samples,) or (n_samples, n_targets)
             Target values.
         """
+        xp, is_array_api = get_namespace(X)
+
+        if is_array_api:
+            svd = xp.linalg.svd
+        else:
+            svd = scipy.linalg.svd
+
         n_samples, n_features = X.shape
-        n_classes = len(self.classes_)
+        n_classes = self.classes_.shape[0]
```

Comment:
It might be nice to allow `len` in the array spec to avoid needing this kind of change. Raised as issue ( https://github.com/data-apis/array-api/issues/481 )

Reply:
This was on purpose and was discussed fairly extensively. The new code is much clearer than the old one, `len` on a >1-D array is an anti-pattern imho.